htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/element/Table.java

```diff
@@ -49,68 +45,70 @@ public Table(WebElement wrappedElement) {
      * @return List with text values of table heading elements.
      */
     public List<String> getHeadingsAsString() {
-        return convert(getHeadings(), toTextValues());
+        return getHeadings().stream()
+                .map(WebElement::getText)
+                .collect(toList());
     }
 
     /**
-     * Returns table cell elements grouped by rows.
+     * Returns table cell elements ({@code <td>}), grouped by rows.
      *
      * @return List where each item is a table row.
      */
     public List<List<WebElement>> getRows() {
-        List<List<WebElement>> rows = new ArrayList<>();
-        List<WebElement> rowElements = getWrappedElement().findElements(By.xpath(".//tr"));
-        for (WebElement rowElement : rowElements) {
-            rows.add(rowElement.findElements(By.xpath(".//td")));
-        }
-        return rows;
+        return getWrappedElement()
+                .findElements(By.cssSelector("tr"))
+                .stream()
+                .map(rowElement -> rowElement.findElements(By.cssSelector("td")))
+                .filter(row -> row.size() > 0) // ignore rows with no <td> tags
+                .collect(toList());
     }
 
     /**
-     * Returns text values of table cell elements grouped by rows.
+     * Returns text values of table cell elements ({@code <td>}), grouped by rows.
      *
      * @return List where each item is text values of a table row.
      */
     public List<List<String>> getRowsAsString() {
-        return convert(getRows(), toListsConvertingEachItem(toTextValues()));
+        return getRows().stream()
+                .map(row -> row.stream()
+                        .map(WebElement::getText)
+                        .collect(toList()))
+                .collect(toList());
     }
 
     /**
-     * Returns table cell elements grouped by columns.
+     * Returns table cell elements ({@code <td>}), grouped by columns.
      *
      * @return List where each item is a table column.
      */
     public List<List<WebElement>> getColumns() {
-        List<List<WebElement>> columns = new ArrayList<>();
-        List<List<WebElement>> rows = getRows();
-
-        if (rows.isEmpty()) {
-            return columns;
-        }
-
-        int columnsNumber = rows.get(0).size();
-        for (int i = 0; i < columnsNumber; i++) {
-            List<WebElement> column = new ArrayList<>();
-            for (List<WebElement> row : rows) {
-                column.add(row.get(i));
-            }
-            columns.add(column);
-        }
-
-        return columns;
+        // assumes subsequent rows have same number of columns as
+        // this first row which contains <td>
+        List<WebElement> firstRow = getWrappedElement().findElements(
+                By.cssSelector("tr:nth-of-type(1) > td"));
```

Comment:
This is true, I will move this to a new method which you can chose which columns you would like, therefore should be faster if the number of columns you need is less than the total number.